src/main/java/de/tum/in/www1/artemis/web/rest/repository/RepositoryProgrammingExerciseParticipationResource.java

```diff
@@ -103,6 +103,28 @@ boolean canAccessRepository(Long participationId) throws IllegalArgumentExceptio
         return super.getFiles(participationId);
     }
 
+    /**
+     * GET /repository/{participationId}/files-change
+     *
+     * Gets the files of the repository and checks whether they were changed during a student participation with respect to the initial template
+     *
+     * @param participationId participation of the student
+     * @return the ResponseEntity with status 200 (OK) and a map of files with the information if they were changed/are new.
+     */
+    @GetMapping(value = "/repository/{participationId}/files-change", produces = MediaType.APPLICATION_JSON_VALUE)
+    @PreAuthorize("hasAnyRole('TA', 'INSTRUCTOR', 'ADMIN')")
+    public ResponseEntity<Map<String, Boolean>> getFilesWithInformationAboutChange(@PathVariable Long participationId) {
+        return super.executeAndCheckForExceptions(() -> {
```

Comment:
How is a permission check performed here?

Reply:
@jpbernius in the `getRepository()` a few checks are done e.g. `canAccessParticipation()` in `ProgrammingExerciseParticipationService.java:187` which basically checks if the user has the correct rights to view the participation.

Or which specific checks do you mean?